### PR TITLE
Replace usage of `.IntoProto().String()` by custom stringer

### PR DESCRIPTION
**Pull request by ValentinTrinque**
Close #5115

### CODE REVIEW

**Review by EVODelavega — DISMISSED** [review]
At first glance, LGTM, but maybe wait for a second review before merging

**Review by jeremyletang — CHANGES_REQUESTED** [review]
A few .IntoProto().String() are left:
```
➜  vega git:(feature/5115-proto-string) ✗ grep -rn '.IntoProto().String()' .
./banking/snapshot.go:175:			e.log.Info("serialiseDeposits:", logging.Int("index", i), logging.String("ID", d.ID), logging.String("deposit", d.Deposit.IntoProto().String()))
./logging/fields.go:171:	return zap.String("order", o.IntoProto().String())
./logging/fields.go:190:	return zap.String(tag, o.IntoProto().String())
```

**Review by wwestgarth — DISMISSED** [review]

**Review by wwestgarth — APPROVED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**
